README.md

```diff
@@ -176,6 +176,27 @@ Sample request after deploying.
 curl https://xxxxxx.execute-api.us-east-1.amazonaws.com/dev/s3 -d '{"message": "testtest"}' -H 'Content-Type:application/json'
 ```
 
+#### Customizing request parameters
+
+Similar to the [SQS](#sqs) support, you can customize the default request parameters `serverless.yml` like so:
+
+```yml
+custom:
+  apiGatewayServiceProxies:
+    - s3:
+        path: /s3
+        method: post
+        action: PutObject
+        bucket:
+          Ref: S3Bucket
+        key: static-key.json
+        cors: true
+
+        requestParameters:
+          # use auto generated object key (will override the static key)
+          'integration.request.path.object': 'context.requestId'
```

Comment:
I'm confused by the configuration here, so we allow a `key` as well as `requestParameter` and let the latter override the first? why not making them mutually exclusive?

Reply:
do you know if we can have other request parameters? maybe for specifying S3 metadata? can they be specified via request params?

Reply:
> I'm confused by the configuration here, so we allow a `key` as well as `requestParameter` and let the latter override the first? why not making them mutually exclusive?

You're right, this is a bit confusing. I thought about making them mutually exclusive, but it requires specifically checking for the 'integration.request.path.object' property in the Joi schema. Let me have another look if I can do it in a non-ugly way

Reply:
> do you know if we can have other request parameters? maybe for specifying S3 metadata? can they be specified via request params?

I'll test if I can set the cache control for example per https://docs.aws.amazon.com/AmazonS3/latest/API/RESTObjectPUT.html

Reply:
thanks, much appreciated. I like the approach, just not keen on having to put in a redundant `key` configuration only to then override it with another thing.